samples/src/main/java/org/opensearch/client/samples/Bulk.java

```diff
@@ -70,21 +72,46 @@ public static void main(String[] args) {
             BulkResponse bulkResponse = client.bulk(bulkReq.build());
             LOGGER.info("Bulk response items: {}", bulkResponse.items().size());
 
-            Query query = Query.of(qb -> qb.match(mb -> mb.field("title").query(fv -> fv.stringValue("Document"))));
-            final SearchRequest.Builder searchReq = new SearchRequest.Builder().allowPartialSearchResults(false)
-                .index(List.of(indexName))
-                .size(10)
-                .source(sc -> sc.fetch(false))
-                .ignoreThrottled(false)
-                .query(query);
-            SearchResponse<IndexData> searchResponse = client.search(searchReq.build(), IndexData.class);
+            // wait for the changes to reflect
+            Thread.sleep(3000);
+            LOGGER.info("Search & bulk update documents");
+
+            SearchResponse<IndexData> searchResponse = search(client, indexName, "title", "Document");
             LOGGER.info("Found {} documents", searchResponse.hits().hits().size());
 
+            for (var hit : searchResponse.hits().hits()) {
+                LOGGER.info("Found {} with score {} and id {}", hit.source(), hit.score(), hit.id());
+                IndexData finalSearchedData = hit.source();
+                finalSearchedData.setText("Updated document");
+                BulkRequest request = new BulkRequest.Builder().operations(
+                    o -> o.update(u -> u.index(indexName).id(hit.id()).document(finalSearchedData))
+                ).build();
+                bulkResponse = client.bulk(request);
+                LOGGER.info("Bulk update response items: {}", bulkResponse.items().size());
+            }
+            // wait for the changes to reflect
+            Thread.sleep(3000);
+            searchResponse = search(client, indexName, "title", "Document");
+
+            for (var hit : searchResponse.hits().hits()) {
+                LOGGER.info("Found {} with score {} and id {}", hit.source(), hit.score(), hit.id());
+            }
             LOGGER.info("Deleting index {}", indexName);
             DeleteIndexRequest deleteIndexRequest = new DeleteIndexRequest.Builder().index(indexName).build();
             client.indices().delete(deleteIndexRequest);
         } catch (Exception e) {
             LOGGER.error("Unexpected exception", e);
         }
     }
+
+    public static SearchResponse<IndexData> search(OpenSearchClient client, String indexName, String field, String value)
```

Comment:
No need, the previous search snippet was just fine

Reply:
Since I'm searching twice in the code (before and after the bulk update), it seems cleaner to extract this into a method.

Reply:
There is no need to search twice - we do have a document id to update

Reply:
the second search was just to show that the update made was reflected, but I can remove it.